### ISSUE TITLE
Lint cache serves stale MD057 verdicts when link target existence changes

### ISSUE THREAD

**Issue by marcodeltongo** [rvben/rumdl#813]
## Summary

The lint cache keys a file's verdict on the file's own content (plus config and rumdl version), but MD057 (existing-relative-links) depends on the existence of OTHER files. When neighboring files change while the source file stays byte-identical, `rumdl check` keeps serving the old verdict, in both directions:

- False positive: a link target is created after a failing run (for example a rebase or branch switch brings the file in). `rumdl check` keeps reporting "Relative link 'X' does not exist" and exits 1 even though the link is now valid. In our repo this blocks pushes from a pre-push hook.
- False negative: a link target is deleted after a passing run. `rumdl check` keeps passing and the broken link goes unreported.

`workspace_index.bin` is rebuilt on each run (its mtime changes), but the per-file verdict is still served from the lint cache, so the refreshed index has no effect on the outcome.

This is adjacent to #640 but distinct: #640 was about the cache-hit fast path dropping suppression data; this one is about the cache key itself not accounting for cross-file state that MD057 depends on.

## Affected versions

Reproduced on 0.2.44 and 0.2.55 (latest at time of writing), macOS arm64 binaries, default cache dir `.rumdl_cache`.

## Reproduction (false positive)

```bash
mkdir /tmp/rumdl-md057-cache && cd /tmp/rumdl-md057-cache
touch .rumdl.toml   # anchor project root so the cache lives here

printf '# A\n\n[see the B document](b.md)\n' > a.md

rumdl check a.md
# a.md:3:22: [MD057] Relative link 'b.md' does not exist
# exit 1 (correct, and warms the cache)

printf '# B\n' > b.md

rumdl check a.md b.md
# a.md:3:22: [MD057] Relative link 'b.md' does not exist
# exit 1 (STALE: b.md exists now)

rumdl check --no-cache a.md b.md
# Success: No issues found in 2 files
# exit 0 (correct)
```

## Reproduction (false negative)

```bash
mkdir /tmp/rumdl-md057-cache-fn && cd /tmp/rumdl-md057-cache-fn
touch .rumdl.toml

printf '# A\n\n[see the B document](b.md)\n' > a.md
printf '# B\n' > b.md

rumdl check a.md b.md
# Success: No issues found in 2 files (correct, warms the cache)

rm b.md

rumdl check a.md
# Success: No issues found in 1 file
# exit 0 (STALE: the link is now broken)

rumdl check --no-cache a.md
# a.md:3:22: [MD057] Relative link 'b.md' does not exist (correct)
```

`rumdl fmt` prints the same stale MD057 warning on a cache hit (its formatter-style exit code is unaffected), so the stale diagnostics also surface in fmt output.

## Expected

A cached verdict should be invalidated, or MD057 re-evaluated, when the existence of the link targets a file references changes. Possible directions:

- include the referenced link targets' existence in the cache key (or in a validity check) for files with relative links, or
- keep cross-file rules like MD057 out of the per-file verdict cache and re-evaluate them against the fresh workspace index on every run.

Our local mitigation is `--no-cache` on the hook invocation (the cache saved us only ~25 ms over ~150 files, so this is cheap). Happy to test a fix; the reproductions above are fully scripted on our side.

**Comment by rvben** [comment]
Fixed in v0.2.56 (f6ae68fb). The cache entry for a file now records a fingerprint of the resolved state of every link target MD057 checked, taken from the workspace index, so a target that appears or disappears while the source file stays byte-identical invalidates the entry in both directions: your false-positive reproduction (target created after a failing run) and the false-negative one (target deleted after a passing run) both re-lint on the next `rumdl check`. Thanks for the precise write-up, the two-direction reproduction made this quick to pin down.
